jlm/llvm/ir/operators/Store.cpp

```diff
@@ -63,6 +97,41 @@ StoreVolatileOperation::copy() const
   return std::unique_ptr<rvsdg::operation>(new StoreVolatileOperation(*this));
 }
 
+[[nodiscard]] size_t
+StoreVolatileOperation::NumMemoryStates() const noexcept
+{
+  // Subtracting I/O state
+  return nresults() - 1;
+}
+
+[[nodiscard]] const StoreVolatileOperation &
+StoreVolatileNode::GetOperation() const noexcept
+{
+  return *util::AssertedCast<const StoreVolatileOperation>(&operation());
+}
+
+[[nodiscard]] StoreNode::MemoryStateInputRange
+StoreVolatileNode::MemoryStateInputs() const noexcept
+{
+  if (NumMemoryStates() == 0)
+  {
+    return { MemoryStateInputIterator(nullptr), MemoryStateInputIterator(nullptr) };
+  }
+
+  return { MemoryStateInputIterator(input(3)), MemoryStateInputIterator(nullptr) };
+}
+
+[[nodiscard]] StoreNode::MemoryStateOutputRange
+StoreVolatileNode::MemoryStateOutputs() const noexcept
+{
+  if (NumMemoryStates())
+  {
+    return { MemoryStateOutputIterator(nullptr), MemoryStateOutputIterator(nullptr) };
```

Comment:
This is backwards, no? If there are memory states, the iterator pair should not start with nullptr?

Reply:
Really great catch!